TUM Campus App/GradeCollectionViewCell.swift

```diff
@@ -11,13 +11,33 @@ import UIKit
 
 class GradeCollectionViewCell: UICollectionViewCell, SingleDataElementPresentable {
     
-    @IBOutlet weak var label1: UILabel!
+    @IBOutlet weak var lectureLabel: UILabel!
+    @IBOutlet weak var semesterLabel: UILabel!
+    @IBOutlet weak var gradeLabel: UILabel!
     
     
     func setElement(_ element: DataElement) {
         
         if let element = element as? Grade {
```

Comment:
Unwrapping every time seems a bit tedious. Perhaps you should add an Associated Type to the protocols and write a wrapper that doesn't have them to be able to hand over the data without knowing the associated type